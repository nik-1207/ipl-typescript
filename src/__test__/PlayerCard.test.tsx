import { render, screen } from "@testing-library/react";
import PlayerCard from "../components/PlayerCard";

it("testing Player card", async () => {
  const teamName = "chennai-super-kings";
  const name = "MS Dhoni";
  const img = "https://iplstatic.s3.amazonaws.com/players/210/1.png";
  const matches = 15;
  const runs = 416;
  const wickets = 0;

  render(
    <PlayerCard
      teamName={teamName}
      img={img}
      name={name}
      matches={matches}
      runs={runs}
      wickets={wickets}
    />
  );
  const card = await screen.findByTestId("player-card");
  expect(card).toMatchSnapshot();
});
