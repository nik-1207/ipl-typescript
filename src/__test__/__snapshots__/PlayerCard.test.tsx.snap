// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Player card 1`] = `
<div
  class="playerCard-0-2-1"
  data-testid="player-card"
>
  <div
    class="chennai-super-kings-0-2-9"
  >
    <img
      alt="MS Dhoni"
      class="chennai-super-kings-0-2-9"
      src="https://iplstatic.s3.amazonaws.com/players/210/1.png"
    />
  </div>
  <div
    class="playerDetail-0-2-2"
  >
    <p
      class="name-0-2-3"
    >
      MS Dhoni
    </p>
    <ul
      class="stats-0-2-4"
    >
      <li
        class="listItem-0-2-5"
      >
        <span
          class="value-0-2-6"
        >
          15
        </span>
        <br />
        <span
          class="label-0-2-7"
        >
          Matches
        </span>
      </li>
      <li
        class="listItem-0-2-5"
      >
        <span
          class="value-0-2-6"
        >
          416
        </span>
        <br />
        <span
          class="label-0-2-7"
        >
          Runs
        </span>
      </li>
      <li
        class="listItem-0-2-5"
      >
        <span
          class="value-0-2-6"
        >
          0
        </span>
        <br />
        <span
          class="label-0-2-7"
        >
          Wickets
        </span>
      </li>
    </ul>
    <button
      class="button-0-2-8"
    >
      View Profile
    </button>
  </div>
</div>
`;
